src/osd/ReplicatedBackend.cc

```diff
@@ -619,16 +619,14 @@ void ReplicatedBackend::submit_transaction(
   op_t->register_on_applied(
     parent->bless_context(
       new C_OSD_OnOpApplied(this, &op)));
-  op_t->register_on_applied(
-    new ObjectStore::C_DeleteTransaction(op_t));
   op_t->register_on_commit(
     parent->bless_context(
       new C_OSD_OnOpCommit(this, &op)));
-
-  list<ObjectStore::Transaction*> tls;
-  tls.push_back(op_t);
+  vector<ObjectStore::Transaction> tls;
+  tls.push_back(std::move(*op_t));
   parent->queue_transactions(tls, op.op);
   delete t;
+  delete op_t;
```

Comment:
it would be nice to get rid of this allocation from RPGTransaction, especially if this is part of main io path. can you make RPGTransaction::t a non-pointer member? then RPGTransaction::get_transaction() would look like:
```
ObjectStore::Transaction&& get_transaction() { return std::move(t); }
```